src/arkode/arkode_lsrkstep.c

```diff
@@ -0,0 +1,1692 @@
+  fprintf(outfile, "LSRKStep: nstsig              = %i\n", step_mem->nstsig);
+  fprintf(outfile, "LSRKStep: stage_max           = %i\n", step_mem->stage_max);
+  fprintf(outfile, "LSRKStep: stage_max_limit     = %i\n",
+          step_mem->stage_max_limit);
+  fprintf(outfile, "LSRKStep: dom_eig_freq        = %i\n",
+          step_mem->dom_eig_freq);
+
+  /* output long integer quantities */
+  fprintf(outfile, "LSRKStep: nfe                 = %li\n", step_mem->nfe);
+  fprintf(outfile, "LSRKStep: dom_eig_nfe         = %li\n",
+          step_mem->dom_eig_nfe);
+  fprintf(outfile, "LSRKStep: num_dom_eig_updates = %li\n",
+          step_mem->num_dom_eig_updates);
+
+  /* output sunrealtype quantities */
+  fprintf(outfile, "LSRKStep: dom_eig             = %f + i%f\n",
+          step_mem->lambdaR, step_mem->lambdaI);
+  fprintf(outfile, "LSRKStep: sprad               = %f\n", step_mem->sprad);
+  fprintf(outfile, "LSRKStep: spr_max             = %f\n", step_mem->spr_max);
+  fprintf(outfile, "LSRKStep: spr_min             = %f\n", step_mem->spr_min);
+  fprintf(outfile, "LSRKStep: dom_eig_sfty        = %f\n",
+          step_mem->dom_eig_sfty);
+
+  /* output sunbooleantype quantities */
+  fprintf(outfile, "LSRKStep: is_ext_dom_eig      = %d\n",
+          step_mem->is_ext_dom_eig);
+  fprintf(outfile, "LSRKStep: new_dom_eig         = %d\n", step_mem->new_dom_eig);
+  fprintf(outfile, "LSRKStep: dom_eig_is_current  = %d\n",
+          step_mem->dom_eig_is_current);
+
+#ifdef SUNDIALS_DEBUG_PRINTVEC
+  /* output vector quantities */
+
+#endif
+}
+
+/*---------------------------------------------------------------
+  lsrkStep_Init:
+
+  This routine is called just prior to performing internal time
+  steps (after all user "set" routines have been called) from
+  within arkInitialSetup.
+
+  With initialization types FIRST_INIT this routine:
+  - sets/checks the ARK Butcher tables to be used
+  - allocates any memory that depends on the number of ARK
+    stages, method order, or solver options
+  - sets the call_fullrhs flag
+
+  With other initialization types, this routine does nothing.
+  ---------------------------------------------------------------*/
+int lsrkStep_Init(ARKodeMem ark_mem, int init_type)
+{
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeLSRKStepMem structure */
+  retval = lsrkStep_AccessStepMem(ark_mem, __func__, &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  /* immediately return if resize or reset */
+  if (init_type == RESIZE_INIT || init_type == RESET_INIT)
+  {
+    return (ARK_SUCCESS);
+  }
+  /* enforce use of arkEwtSmallReal if using a fixed step size
+     and an internal error weight function */
+  if (ark_mem->fixedstep && !ark_mem->user_efun)
+  {
+    ark_mem->user_efun = SUNFALSE;
+    ark_mem->efun      = arkEwtSetSmallReal;
+    ark_mem->e_data    = ark_mem;
+  }
+
+  /* Set the default stepper */
+  if (ark_mem->step == lsrkStep_TakeStepRKC)
```

Comment:
This is used as a flag only to set the default stepper. This so called 'flag" means `LSRKStepSetMethod` is not called with a different method type. Then, it will set the default method `ARKODE_LSRK_RKC_2`.

Reply:
As originally written, this omitted the "default" SSP method.  Now that I've added that below, it covers all cases.